### PR TITLE
Espnet2 mucs_subtask2

### PR DESCRIPTION
**Pull request by bloodraven66**
- [x] Integrate end to end (data.sh disconnected currently)
- [x] train espnet2 model (conformer trained, transformer trained on 1 language, one more left)

### CODE REVIEW

**Review by ftshijt — COMMENTED** [review]
The current status seems fine. Please keep on the remaining part. Many thanks!

**Comment by bloodraven66** [comment]
@ftshijt that should be it regarding the codes. Please let me know if I'm missing anything!

File: egs/mucs21_subtask2/asr1/local/download_data.sh [hunk 1]
```diff
@@ -35,22 +35,35 @@ declare -A trainset
 trainset['hi-en']='http://www.ee.iisc.ac.in/new/people/faculty/prasantg/downloads/Hindi-English_train.tar.gz'
 trainset['bn-en']='http://www.ee.iisc.ac.in/new/people/faculty/prasantg/downloads/Bengali-English_train.tar.gz'
 
+declare -A valset
+# testset['hi']='http://www.ee.iisc.ac.in/new/people/faculty/prasantg/downloads/Hindi_test.tar.gz'
+# testset['mr']='http://www.ee.iisc.ac.in/new/people/faculty/prasantg/downloads/Marathi_test.tar.gz'
+# testset['or']='http://www.ee.iisc.ac.in/new/people/faculty/prasantg/downloads/Odia_test.tar.gz'
```
Comment (ftshijt) [inline egs/mucs21_subtask2/asr1/local/download_data.sh:41]: are these lines necessary?

File: egs2/mucs21_subtask2/asr1/local/data.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
+lang=hi-en
```
Comment (ftshijt) [inline egs2/mucs21_subtask2/asr1/local/data.sh:1]: Could you also add log and interaction with `db.sh` ?

**Review by ftshijt — COMMENTED** [review]
I notice some files are highly correlated with `egs`. In that case, I would recommend to use symlink from there instead of copying.

File: egs2/mucs21_subtask2/asr1/run.sh [hunk 3]
```diff
@@ -15,6 +15,7 @@ lm_config_=conf/lm.yaml
     --lang hi-en \
     --ngpu 2 \
     --use_lm true \
+    --skip_data_prep true \
```
Comment (ftshijt) [inline egs2/mucs21_subtask2/asr1/run.sh]: I think we should not make it default, right?

**Comment by ftshijt** [comment]
Please fix the conflict~ Other codebases look cool now. Many thanks!

**Review by pengchengguo — APPROVED** [review]

File: egs2/mucs21_subtask2/asr1/conf/specaug.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+process:
```
Comment (sw005320) [inline egs2/mucs21_subtask2/asr1/conf/specaug.yaml:1]: ditto.
espnet2 usually put the specaug parameters in the training config file.
Please do it, and run the experiment, again.

File: egs2/mucs21_subtask2/asr1/run.sh [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+asr_config=conf/train.yaml
+lm_config_=conf/lm.yaml
+
+./asr.sh \
+    --lang $lang \
+    --ngpu 2 \
+    --expdir exp \
+    --local_data_opts $lang \
+    --use_lm true \
+    --skip_upload false \
```
Comment (sw005320) [inline egs2/mucs21_subtask2/asr1/run.sh:23]: This should leave it as default (remove this line)
The main user would not upload the model.

File: egs2/mucs21_subtask2/asr1/run.sh [hunk 6]
```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+# Set bash to 'debug' mode, it will exit on :
+# -e 'error', -u 'undefined variable', -o ... 'error in pipeline', -x 'print commands',
+set -e
+set -u
+set -o pipefail
+
+lang=hi-en
```
Comment (sw005320) [inline egs2/mucs21_subtask2/asr1/run.sh:8]: add a comment that we can use hi-en, or, and bn-en

**Review by sw005320 — COMMENTED** [review]
- Please fix the conflict `egs2/TEMPLATE/asr1/db.sh`
- basically same comment as subtask2. After you fix some points, I'll merge this PR
- please make a follow up PR to add a result and corresponding model.

**Comment by sw005320** [comment]
Thanks, @bloodraven66!
And please also update a result and upload a model, later!

**Pull request merged**